src/coreComponents/constitutive/contact/ContactRelationBase.hpp

```diff
@@ -106,6 +106,38 @@ class ContactRelationBase : public ConstitutiveBase
 
   inline real64 apertureTolerance() const { return m_apertureTolerance; }
 
+  virtual void computeTraction( R1Tensor const & dispJump,
+                                R1Tensor & tractionVector ) const;
+
+  void addPressureToTraction ( real64 const & pressure,
+                               R1Tensor & tractionVector ) const
+  {
+    tractionVector[0] += pressure;
+  }
+
+  void dTraction_dPressure( real64 & dTdpf,
+                            bool const open ) const
+  {
+    dTdpf = open ? 1.0 : 0.0;
+  }
+
+  virtual void dTraction_dJump( R1Tensor const & dispJump,
+                                arraySlice2d< real64 > const & dTdw ) const
+  {
+
+    dTdw( 0, 0 ) = dispJump[0] >=0 ? 0.0 : m_penaltyStiffness;
+
+    // all the others are zeros
+    dTdw( 0, 1 ) = 0.0;
+    dTdw( 0, 2 ) = 0.0;
+    dTdw( 1, 0 ) = 0.0;
+    dTdw( 1, 1 ) = 0.0;
+    dTdw( 1, 2 ) = 0.0;
+    dTdw( 2, 0 ) = 0.0;
+    dTdw( 2, 1 ) = 0.0;
+    dTdw( 2, 2 ) = 0.0;
+  }
```

Comment:
For the frictionless contact case the traction is defined as:

```
traction[0] = m_penaltyStiffness * jump[0] - p ;
traction[1] = 0 ;
traction[2] = 0;
```
I think the sign convention used in the computation of the derivatives should be consistified.